### PR TITLE
return us-east-1 when LocationConstraint is null

### PR DESCRIPTION
**Pull request by jhnlsn**
#1720

### CODE REVIEW

**Comment by ewbankkit** [comment]
How about adding a function
```
def get_region(b):
    return b.get('Location', {}).get('LocationConstraint') or 'us-east-1'
```
and calling from:
* https://github.com/capitalone/cloud-custodian/blob/c3c132de1896e1fb68ef53414eb5588d1ad43481/c7n/resources/s3.py#L461
* https://github.com/capitalone/cloud-custodian/blob/c3c132de1896e1fb68ef53414eb5588d1ad43481/c7n/resources/s3.py#L1027
* https://github.com/capitalone/cloud-custodian/blob/c3c132de1896e1fb68ef53414eb5588d1ad43481/c7n/resources/s3.py#L1049

**Comment by jhnlsn** [comment]
I did actually think about that!

**Comment by kapilt** [comment]
this conversion applies to eu as well

**Comment by jhnlsn** [comment]
@kapilt I believe that only buckets created in the Virginia location will return a Null location constraint.  Is that what you mean by eu?

File: c7n/resources/s3.py [hunk 1]
```diff
@@ -495,6 +491,11 @@ def modify_bucket_tags(session_factory, buckets, add_tags=(), remove_tags=()):
             continue
 
 
+def get_region(b):
+    return b.get('Location', {
+        'LocationConstraint': 'us-east-1'})['LocationConstraint'] or 'us-east-1'
```
Comment (kapilt) [inline c7n/resources/s3.py]: there's another oddity that should be accounted for here, namely for a value of 'EU' it should translate to 'eu-west-1'

Comment (jhnlsn) [reply c7n/resources/s3.py]: we all love you aws 👍

**Pull request merged**